presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -149,27 +153,31 @@
     private final HiveCluster clientProvider;
     private final Function<Exception, Exception> exceptionMapper;
     private final boolean impersonationEnabled;
+    private final HiveMetastoreAuthentication metastoreAuthentication;
 
     @Inject
-    public ThriftHiveMetastore(HiveCluster hiveCluster, MetastoreClientConfig config)
+    public ThriftHiveMetastore(HiveCluster hiveCluster, MetastoreClientConfig config, HiveMetastoreAuthentication metastoreAuthentication)
     {
         this(
                 hiveCluster,
                 new ThriftHiveMetastoreStats(),
                 identity(),
-                requireNonNull(config, "config is null").isMetastoreImpersonationEnabled());
+                requireNonNull(config, "config is null").isMetastoreImpersonationEnabled(),
+                metastoreAuthentication);
     }
 
     public ThriftHiveMetastore(
             HiveCluster hiveCluster,
             ThriftHiveMetastoreStats stats,
             Function<Exception, Exception> exceptionMapper,
-            boolean impersonationEnabled)
+            boolean impersonationEnabled,
+            HiveMetastoreAuthentication metastoreAuthentication)
```

Comment:
I see we already have a config `HiveMetastoreAuthenticationType` that tells us if authentication is enabled or not. We should use that here. Instead of passing the `HiveMetastoreAuthentication` and comparing the class name, i think we should pass the `HiveMetastoreAuthenticationType` here